src/array.cr

```diff
@@ -1216,6 +1216,12 @@ class Array(T)
   #
   # Accepts an optional *offset* parameter, which tells it to start counting
   # from there.
+  #
+  # ```
+  # gems = ["crystal", "pearl", "diamond"]
+  # results = gems.map_with_index { |e, i| "#{i}: #{e}" }
```

Comment:
```suggestion
  # results = gems.map_with_index { |gem, idx| "#{idx}: #{gem}" }
```

Reply:
I think `i` is fine - and `idx` doesn't have any noticable benefit. But `gem` is a good idea.

Reply:
I updated `gem` but not `idx`.  I think that is fine but let me all know what you think.